lib/compress.js

```diff
@@ -1731,6 +1731,7 @@ merge(Compressor.prototype, {
                 var val = {};
                 for (var i = 0, len = this.properties.length; i < len; i++) {
                     var prop = this.properties[i];
+                    if (prop instanceof AST_Expansion) return this;
```

Comment:
@alexlamsl Regarding `may_throw_on_access` and `drop_side_effect_free` - can you show an example of object rest/spread code that would break with each of them? I am unable to create such a case:
```
$ echo 'var a = 1; ({ ...foo(), ...a, ...g, x: bar() });' | bin/uglifyjs -cm --toplevel
foo(),g,bar();
```